R/assert_cran_url.R

```diff
@@ -0,0 +1,52 @@
+#' @title Validate the URL of a package on CRAN.
+#' @export
+#' @keywords internal
+#' @description If the package is on CRAN, validate that the contributed URL
+#'   appears in the CRAN-hosted DESCRIPTION file.
+#' @return A character string if there is a problem with the package entry,
+#'   otherwise `NULL` if there are no issues.
+#' @param name Character of length 1, contributed name of the package.
+#' @param url Character of length 1, contributed URL of the package.
+assert_cran_url <- function(name, url) {
+  result <- try(
+    pkgsearch::cran_package(name = name),
+    silent = TRUE
+  )
+  if (inherits(result, "try-error")) {
+    return(invisible())
+  }
+  package <- result[["Package"]]
+  if (!identical(package, name)) {
+    return(invisible())
+  }
```

Comment:
Cannot return NULL here if the name is not identical.
Not needed any more as the previous check will error if the package is not on CRAN.